### PR TITLE
Loki: Allow alert headers to be forwarded

### PR DESCRIPTION
**Pull request by svennergr**
**What is this feature?**

This is a followup of https://github.com/grafana/grafana/pull/90344. Headers that should be forwarded need to be prefixed with `http_` and the http client's `ForwardHTTPHeaders` is enabled.

### CODE REVIEW

**Review by matyax — APPROVED** [review]

File: pkg/services/ngalert/eval/eval.go [hunk 1]
```diff
@@ -334,7 +334,7 @@ func buildDatasourceHeaders(ctx context.Context, metadata map[string]string) map
 	headers := make(map[string]string, len(metadata)+3)
 
 	for key, value := range metadata {
-		headers[fmt.Sprintf("X-Rule-%s", key)] = url.QueryEscape(value)
+		headers[fmt.Sprintf("http_X-Rule-%s", key)] = url.QueryEscape(value)
 	}
```
Comment (yuri-tceretian) [inline pkg/services/ngalert/eval/eval.go:338]: this changes how it is sent to other data sources, doesn't it?

Comment (svennergr) [reply pkg/services/ngalert/eval/eval.go:338]: Yea, it is the standard per se. This part in the plugin-sdk will pull headers with `http_` prefixed and add them: https://github.com/grafana/grafana-plugin-sdk-go/blob/c5aa822be58253fd954e921aec00241833d9ca4c/backend/http_headers.go#L74-L77

**Review by yuri-tceretian — APPROVED** [review]
LGTM

**Pull request merged**